src/client/app/components/IntervalControlsComponent.tsx

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as moment from 'moment';
+import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { FormFeedback, FormGroup, Input, Label } from 'reactstrap';
+import { useAppDispatch, useAppSelector } from '../redux/reduxHooks';
+import { selectChartToRender, graphSlice, selectWidthDays, selectComparePeriod } from '../redux/slices/graphSlice';
+import { ChartTypes } from '../types/redux/graph';
+import { ComparePeriod } from '../utils/calculateCompare';
+import translate from '../utils/translate';
+import TooltipMarkerComponent from './TooltipMarkerComponent';
+
+/**
+ * @returns Interval controls for the bar, map, and compare pages
+ */
+export default function IntervalControlsComponent() {
+	const dispatch = useAppDispatch();
+	const chartType = useAppSelector(selectChartToRender);
+
+	// The min/max days allowed for user selection
+	const MIN_DAYS = 1;
+	const MAX_DAYS = 366;
+	// Special value if custom input for standard menu.
+	const CUSTOM_INPUT = '-99';
+
+	// This is the current interval for the bar and map graphics.
+	const duration = useAppSelector(selectWidthDays);
+	// This is the current compare period for graphic
+	const comparePeriod = chartType === ChartTypes.compare ? useAppSelector(selectComparePeriod) : undefined;
+
+	// Holds the value of standard duration choices used for bar and map, decoupled from custom.
+	const [days, setDays] = React.useState<string>(duration.asDays().toString());
+	// Holds the value during custom duration input for bar and map, so only update graphic
+	// when done entering and separate from standard choices.
+	const [daysCustom, setDaysCustom] = React.useState<number>(duration.asDays());
+	// True if custom duration input for bar or map is active.
+	const [showCustomDuration, setShowCustomDuration] = React.useState<boolean>(false);
+	// Define a flag to track if custom input is actively being used
+	const [isCustomInput, setIsCustomInput] = React.useState<boolean>(false);
+
+	// Keeps react-level state, and redux state in sync.
+	// Two different layers in state may differ especially when externally updated (chart link, history buttons.)
+	React.useEffect(() => {
+		// If user is in custom input mode, don't reset to standard options
+		if (!isCustomInput) {
+			const isCustom = !['1', '7', '28'].includes(duration.asDays().toString());
```

Comment:
To fully integrate looping over the enum for duration, what if this line is replaced with:
```js
			const durationValues = Object.values(ComparePeriod) as string[];
			const isCustom = !(durationValues.includes(duration.asDays().toString()));
```
What do you think?

Reply:
That works.